interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java

```diff
@@ -1738,14 +1741,19 @@ public void performSoakTest(
       int soakRequestSize,
       int soakResponseSize)
       throws Exception {
-    int iterationsDone = 0;
-    int totalFailures = 0;
+    // int iterationsDone = 0;
+    // int totalFailures = 0;
+    AtomicInteger iterationsDone = new AtomicInteger(0);
+    AtomicInteger totalFailures = new AtomicInteger(0);
     Histogram latencies = new Histogram(4 /* number of significant value digits */);
     long startNs = System.nanoTime();
     ManagedChannel soakChannel = createChannel();
     TestServiceGrpc.TestServiceBlockingStub soakStub = TestServiceGrpc
         .newBlockingStub(soakChannel)
         .withInterceptors(recordClientCallInterceptor(clientCallCapture));
+    List<Thread> threads = new ArrayList<>();
+    // Only allow up to 10 threads to run concurrently
+    Semaphore semaphore = new Semaphore(10);
     for (int i = 0; i < soakIterations; i++) {
```

Comment:
Rather than hardcoding 10 let's make parallelism configurable with a flag.

We can add a new flag, something like `--soak_num_threads`, where the value of the flag determines the number of number of threads to independently run the soak test loop on.

The flag can default to `1`.

Reply:
- I have updated the code based on your comments. Currently, I am implementing a uniform distribution for the soak iterations across different threads. However, there are several other approaches to distribute the soak iterations dynamically, such as round robin or load balancing. Please let me know if you would prefer a different implementation.
- Also I tried to run "tools/distrib/sanitize.sh" to format the code, however i could not find any sanitize.sh file from the root directory. Any hint?
- I may also need to modify the related .py files. I will update those files and submit a change list (CL) for review afterward.